packages/components/src/components/pin-code/pin-code.tsx

```diff
@@ -0,0 +1,314 @@
+              autoComplete="one-time-code"
+              pattern="\d*"
+              inputMode="numeric" // get numeric keyboard on mobile
+              value={this.value[index]}
+              disabled={this.disabled}
+              required={this.required}
+              ref={(el) => this.pinCodeElements.push(el)}
+            />
+          ))}
+        </div>
+        {hasMessage(this.host, this.message, this.state) && (
+          <StateMessage id="state-message" state={this.state} message={this.message} theme="light" host={this.host} />
+        )}
+      </Host>
+    );
+  }
+
+  private onInput = (
+    e: InputEvent & {
+      target: HTMLInputElement;
+    }
+  ): void => {
+    // needed to update value on auto-complete via keyboard suggestion
+    const { target } = e;
+    if (target.value?.length >= this.length) {
+      const optimizedValue = getOptimizedValue(target.value, this.length);
+      this.value = optimizedValue.split('');
+      this.updateValue();
+      this.focusFirstEmptyOrLastElement();
+    }
+  };
+
+  private onKeyDown = (
+    e: KeyboardEvent & {
+      target: HTMLInputElement & { previousElementSibling: HTMLInputElement; nextElementSibling: HTMLInputElement };
+    }
+  ): void => {
+    const {
+      key,
+      target,
+      target: { previousElementSibling, nextElementSibling },
+    } = e;
+    // prevent default for disabled or loading, but do not impede tab key
+    if (isDisabledOrLoading(this.disabled, this.loading) && key !== 'Tab') {
+      e.preventDefault();
+    } // if input is valid overwrite old value
+    else if (inputIsSingleDigit(key)) {
+      e.preventDefault();
+      target.value = key;
+      this.value = getArrayOfInputValues(this.pinCodeElements);
+      this.updateValue();
+
+      if (nextElementSibling) {
+        nextElementSibling.focus();
+      }
+    } // handle alphanumeric keys, allow copy/paste shortcut
+    else if (key?.length === 1 && !(e.ctrlKey || e.metaKey)) {
+      e.preventDefault();
+    } // handle backspace and delete
+    else if (key === 'Backspace' || key === 'Delete') {
+      // transfer focus backward/forward, if the input value is empty
+      if (!target.value) {
+        e.preventDefault();
+        if (key === 'Backspace' && previousElementSibling) {
+          previousElementSibling.value = '';
+          previousElementSibling.focus();
+        } else if (key === 'Delete' && nextElementSibling) {
+          nextElementSibling.value = '';
+          nextElementSibling.focus();
+        }
+      }
+      target.value = '';
+      this.pinCodeElements.map((pinCodeElement) => {
+        if (pinCodeElement === e.target) {
+          pinCodeElement.value = '';
+        }
+        return pinCodeElement;
+      });
+
+      this.value = getArrayOfInputValues(this.pinCodeElements);
+      this.updateValue();
+    } // support native submit behavior
+    else if (key === 'Enter') {
+      if (isWithinForm) {
+        handleButtonEvent(
```

Comment:
Can a native form with a native input and without a `button type="submit"` really be submitted like that?

Reply:
The `button type="submit"` is needed